components/emoji_picker/emoji_picker_overlay.tsx

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Overlay} from 'react-bootstrap';
+import memoize from 'memoize-one';
+
+import {Emoji} from 'mattermost-redux/types/emojis';
+
+import {popOverOverlayPosition} from 'utils/position_utils';
+import {Constants} from 'utils/constants';
+
+import EmojiPickerTabs from './emoji_picker_tabs';
+
+export enum PositionTypes {
+    LEFT ='left',
+    RIGHT = 'right',
+}
+
+interface Props {
+    show: boolean;
+    container?: () => void;
+    target: () => void;
+    onEmojiClick: (emoji: Emoji) => void;
+    onGifClick?: (gif: string) => void;
+
+    onEmojiClose?: (event?: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+    onHide: () => void;
+    topOffset: number;
+    rightOffset?: number;
+    leftOffset?: number;
+    spaceRequiredAbove?: number;
+    spaceRequiredBelow?: number;
+    enableGifPicker?: boolean;
+    defaultHorizontalPosition?: PositionTypes;
+}
+
+// An emoji picker in the center channel is contained within the post list, so it needs space
+// above for the channel header and below for the post textbox
+const CENTER_SPACE_REQUIRED_ABOVE = 476;
+const CENTER_SPACE_REQUIRED_BELOW = 497;
+
+// An emoji picker in the RHS isn't constrained by the RHS, so it just needs space to fit
+// the emoji picker itself
+// const RHS_SPACE_REQUIRED_ABOVE = 420;
+// const RHS_SPACE_REQUIRED_BELOW = 420;
```

Comment:
Related to the comment in `post_reaction.tsx`:

It would be better to define an exported constant here like this

```suggestion
type Location = 'RHS' | 'CENTER';

export const SpaceRequirement: Record<Location, {ABOVE: number; BELOW: number}> = {

    // An emoji picker in the center channel is contained within the post list, so it needs space
    // above for the channel header and below for the post textbox
    CENTER: {
        ABOVE: 476,
        BELOW: 497,
    },

    // An emoji picker in the RHS isn't constrained by the RHS, so it just needs space to fit
    // the emoji picker itself
    RHS: {
        ABOVE: 420,
        BELOW: 420,
    },
}
```

a further improvement would be to just pass in a location of either `RHS` or `CENTER` and get the correct values from the defined constant, but this is probably out of scope for this PR

```
type Location = 'RHS' | 'CENTER';

const SpaceRequirement: Record<Location, {ABOVE: number; BELOW: number}> = {
    CENTER: {
        ABOVE: 476,
        BELOW: 497,
    },
    RHS: {
        ABOVE: 420,
        BELOW: 420,
    },
};

...

// use it like this

popOverOverlayPosition(targetBounds, window.innerHeight, SpaceRequirement[props.location].ABOVE, SpaceRequirement[props.location].BELOW, defaultHorizontalPosition)
```